ecommerce/frontend/src/components/search/SearchSidePanel.js

```diff
@@ -0,0 +1,123 @@
+        },
+        sm: {
+            span: 8,
+        },
+    },
+    wrapperCol: {
+        xs: {
+            span: 24,
+        },
+        sm: {
+            span: 16,
+        },
+    },
+}
+const tailFormItemLayout = {
+    wrapperCol: {
+        xs: {
+            span: 24,
+            offset: 0,
+        },
+        sm: {
+            span: 16,
+            offset: 8,
+        },
+    },
+}
+
+export const SearchSidePanel = ({ initialValues = {}, onSubmit = () => {} }) => {
+    const [form] = Form.useForm()
+    const [category, setCategory] = useState(initialValues?.category ?? 'electronics')
+
+    const onSelectCategory = category => {
+        setCategory(category)
+        form.setFieldsValue({ subcategory: null })
+    }
+
+    const onFinish = values => {
+        console.log('Search side panel form:', values)
+        onSubmit(values)
+    }
+    const onResetClick = () => {
+        form.resetFields()
+    }
+
+    return (
+        <Form {...formItemLayout} form={form} initialValues={initialValues} onFinish={onFinish}>
+            <Form.Item name="category" label="Category">
+                <Select onSelect={onSelectCategory} placeholder="Category...">
+                    {Object.entries(categories).map(([key, value]) => (
+                        <Select.Option key={key} value={key}>
+                            {value}
+                        </Select.Option>
+                    ))}
+                </Select>
+            </Form.Item>
+            <Form.Item name="subcategory" label="Subcategory">
+                <Select placeholder="Subcategory..." allowClear>
+                    {Object.entries(subcategories[category]).map(([key, value]) => (
+                        <Select.Option key={key} value={key}>
+                            {value}
+                        </Select.Option>
+                    ))}
+                </Select>
+            </Form.Item>
+            <Form.Item name="vendor" label="Vendor">
+                <Select placeholder="Vendor..." allowClear>
+                    {Object.entries(vendors).map(([key, value]) => (
+                        <Select.Option key={key} value={key}>
+                            {value}
+                        </Select.Option>
+                    ))}
+                </Select>
+            </Form.Item>
+            <Form.Item name="brand" label="Brand">
+                <Select placeholder="Brand..." allowClear>
+                    {Object.entries(brands).map(([key, value]) => (
+                        <Select.Option key={key} value={key}>
+                            {value}
+                        </Select.Option>
+                    ))}
+                </Select>
+            </Form.Item>
+            <Form.Item name="priceRange" label="Price range">
+                <Slider range />
+            </Form.Item>
+            <Form.Item name="rating" label="Rating">
+                <Rate allowHalf allowClear />
+            </Form.Item>
```

Comment:
Although original Slider in the antd page is colored blue, my browser renders it as white. 
Also, Rate component is very hard to see. Lets give default value as 5 stars to make it more visible. 

![image](https://user-images.githubusercontent.com/44057640/99707166-2564bd80-2aad-11eb-8015-8a5e08680858.png)
![image](https://user-images.githubusercontent.com/44057640/99707219-331a4300-2aad-11eb-9365-8efcfa612be7.png)

Reply:
Also, Slider default value should be [0 to Max](currently [0-0] by default since we include every item.

Reply:
Oh, you are getting the default values from the query-string parse method. Then, they start with empty default values.

Reply:
I am sorry, I was a couple commit behind you. You added some background and all. It looks better with that purple color

Reply:
I agree with Abdullah about the Rating component. The default should be 5 stars because it just looks more appealing.